In [ ]:
import numpy as np
import pandas as pd

df = pd.read_csv("/content/Structured_whole_dataframe.csv")
df = df.drop("Unnamed: 0", axis=1)
df

doc_id            sentence_id  \
0       DDI-MedLine.d180    DDI-MedLine.d180.s1   
1       DDI-MedLine.d180   DDI-MedLine.d180.s10   
2       DDI-MedLine.d180   DDI-MedLine.d180.s10   
3       DDI-MedLine.d180   DDI-MedLine.d180.s10   
4       DDI-MedLine.d180   DDI-MedLine.d180.s10   
...                  ...                    ...   
33503  DDI-DrugBank.d567  DDI-DrugBank.d567.s22   
33504  DDI-DrugBank.d567  DDI-DrugBank.d567.s22   
33505  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   
33506  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   
33507  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   

                                           sentence_text  \
0      The 3-Deazaneplanocin A (DZNep), one of S-aden...   
1      We also found that Bcl-2 was overexpressed in ...   
2      We also found that Bcl-2 was overexpressed in ...   
3      We also found that Bcl-2 was overexpressed in ...   
4      We also found that Bcl-2 was overexpressed in ...   
...                                                  ...   
33503  Oral Hypoglycemic Agents: In pharmacokinetic s...   
33504  Oral Hypoglycemic Agents: In pharmacokinetic s...   
33505  May interact with addictive medications, espec...   
33506  May interact with addictive medications, espec...   
33507  May interact with addictive medications, espec...   

                                        e1_text e1_type e1_charOffset  \
0                           3-Deazaneplanocin A  drug_n          4-22   
1                                         DZNep  drug_n         46-50   
2                                         DZNep  drug_n         46-50   
3                                         DZNep  drug_n         46-50   
4                                         DZNep  drug_n         92-96   
...                                         ...     ...           ...   
33503                                   MEVACOR   brand         56-62   
33504                                 glipizide    drug       163-171   
33505  central nervous system (CNS) depressants   group         52-91   
33506  central nervous system (CNS) depressants   group         52-91   
33507                                   alcohol    drug       185-191   

              e2_text e2_type e2_charOffset    ddi            type  \
0               DZNep  drug_n         25-29  False  No interaction   
1               DZNep  drug_n         92-96  False  No interaction   
2             ABT-737  drug_n       102-108  False  No interaction   
3               DZNep  drug_n       195-199  False  No interaction   
4             ABT-737  drug_n       102-108   True          effect   
...               ...     ...           ...    ...             ...   
33503  chlorpropamide    drug       181-194  False  No interaction   
33504  chlorpropamide    drug       181-194  False  No interaction   
33505         alcohol    drug       185-191  False  No interaction   
33506      ethinamate    drug       320-329  False  No interaction   
33507      ethinamate    drug       320-329  False  No interaction   

                                                source_x  
0      DDICorpus/Test/Test for DDI Extraction task/Me...  
1      DDICorpus/Test/Test for DDI Extraction task/Me...  
2      DDICorpus/Test/Test for DDI Extraction task/Me...  
3      DDICorpus/Test/Test for DDI Extraction task/Me...  
4      DDICorpus/Test/Test for DDI Extraction task/Me...  
...                                                  ...  
33503        DDICorpus/Train/DrugBank/Lovastatin_ddi.xml  
33504        DDICorpus/Train/DrugBank/Lovastatin_ddi.xml  
33505        DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  
33506        DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  
33507        DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  

[33508 rows x 12 columns]

In [ ]:
df.columns

Index(['doc_id', 'sentence_id', 'sentence_text', 'e1_text', 'e1_type',
       'e1_charOffset', 'e2_text', 'e2_type', 'e2_charOffset', 'ddi', 'type',
       'source_x'],
      dtype='object')

# Number of documents in corpus (Training + Test)

In [ ]:
len(df["doc_id"].unique())

815

# Number of sentences in given 815 documents (Training + Test)

In [ ]:
len(df["sentence_id"].unique())

4579

# Number of documents in Training Set

In [ ]:
train_set = df[df["source_x"].str.startswith("DDICorpus/Train")]
train_set

doc_id            sentence_id  \
5716     DDI-MedLine.d69     DDI-MedLine.d69.s0   
5717     DDI-MedLine.d69     DDI-MedLine.d69.s0   
5718     DDI-MedLine.d69     DDI-MedLine.d69.s0   
5719     DDI-MedLine.d69     DDI-MedLine.d69.s1   
5720     DDI-MedLine.d69     DDI-MedLine.d69.s1   
...                  ...                    ...   
33503  DDI-DrugBank.d567  DDI-DrugBank.d567.s22   
33504  DDI-DrugBank.d567  DDI-DrugBank.d567.s22   
33505  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   
33506  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   
33507  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   

                                           sentence_text  \
5716   Differential regulation of tyrosine phosphoryl...   
5717   Differential regulation of tyrosine phosphoryl...   
5718   Differential regulation of tyrosine phosphoryl...   
5719   The homodimeric disintegrin contortrostatin wa...   
5720   The homodimeric disintegrin contortrostatin wa...   
...                                                  ...   
33503  Oral Hypoglycemic Agents: In pharmacokinetic s...   
33504  Oral Hypoglycemic Agents: In pharmacokinetic s...   
33505  May interact with addictive medications, espec...   
33506  May interact with addictive medications, espec...   
33507  May interact with addictive medications, espec...   

                                        e1_text e1_type e1_charOffset  \
5716                            contortrostatin  drug_n         70-84   
5717                            contortrostatin  drug_n         70-84   
5718                                 echistatin  drug_n       141-150   
5719                            contortrostatin  drug_n         28-42   
5720                            contortrostatin  drug_n         28-42   
...                                         ...     ...           ...   
33503                                   MEVACOR   brand         56-62   
33504                                 glipizide    drug       163-171   
33505  central nervous system (CNS) depressants   group         52-91   
33506  central nervous system (CNS) depressants   group         52-91   
33507                                   alcohol    drug       185-191   

              e2_text e2_type e2_charOffset    ddi            type  \
5716       echistatin  drug_n       141-150  False  No interaction   
5717       flavoridin  drug_n       156-165  False  No interaction   
5718       flavoridin  drug_n       156-165  False  No interaction   
5719       echistatin  drug_n        96-105  False  No interaction   
5720       flavoridin  drug_n       111-120  False  No interaction   
...               ...     ...           ...    ...             ...   
33503  chlorpropamide    drug       181-194  False  No interaction   
33504  chlorpropamide    drug       181-194  False  No interaction   
33505         alcohol    drug       185-191  False  No interaction   
33506      ethinamate    drug       320-329  False  No interaction   
33507      ethinamate    drug       320-329  False  No interaction   

                                          source_x  
5716          DDICorpus/Train/MedLine/10978746.xml  
5717          DDICorpus/Train/MedLine/10978746.xml  
5718          DDICorpus/Train/MedLine/10978746.xml  
5719          DDICorpus/Train/MedLine/10978746.xml  
5720          DDICorpus/Train/MedLine/10978746.xml  
...                                            ...  
33503  DDICorpus/Train/DrugBank/Lovastatin_ddi.xml  
33504  DDICorpus/Train/DrugBank/Lovastatin_ddi.xml  
33505  DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  
33506  DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  
33507  DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  

[27792 rows x 12 columns]

In [ ]:
len(train_set["doc_id"].unique())

624

# Number of Documents in DrugBank Training Set

In [ ]:
train_set_drugbank = df[df["source_x"].str.startswith("DDICorpus/Train/DrugBank")]
train_set_drugbank

doc_id            sentence_id  \
7503   DDI-DrugBank.d481   DDI-DrugBank.d481.s0   
7504   DDI-DrugBank.d419   DDI-DrugBank.d419.s0   
7505   DDI-DrugBank.d419   DDI-DrugBank.d419.s0   
7506   DDI-DrugBank.d419   DDI-DrugBank.d419.s0   
7507   DDI-DrugBank.d419   DDI-DrugBank.d419.s1   
...                  ...                    ...   
33503  DDI-DrugBank.d567  DDI-DrugBank.d567.s22   
33504  DDI-DrugBank.d567  DDI-DrugBank.d567.s22   
33505  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   
33506  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   
33507  DDI-DrugBank.d376   DDI-DrugBank.d376.s0   

                                           sentence_text  \
7503   Milk, milk products, and calcium-rich foods or...   
7504   The concurrent administration of allopurinol a...   
7505   The concurrent administration of allopurinol a...   
7506   The concurrent administration of allopurinol a...   
7507   It is not known whether this potentiation of a...   
...                                                  ...   
33503  Oral Hypoglycemic Agents: In pharmacokinetic s...   
33504  Oral Hypoglycemic Agents: In pharmacokinetic s...   
33505  May interact with addictive medications, espec...   
33506  May interact with addictive medications, espec...   
33507  May interact with addictive medications, espec...   

                                        e1_text e1_type e1_charOffset  \
7503                                    calcium    drug         25-31   
7504                                allopurinol    drug         33-43   
7505                                allopurinol    drug         33-43   
7506                                 ampicillin    drug         49-58   
7507                                 ampicillin    drug         45-54   
...                                         ...     ...           ...   
33503                                   MEVACOR   brand         56-62   
33504                                 glipizide    drug       163-171   
33505  central nervous system (CNS) depressants   group         52-91   
33506  central nervous system (CNS) depressants   group         52-91   
33507                                   alcohol    drug       185-191   

              e2_text e2_type e2_charOffset    ddi            type  \
7503            EMCYT   brand         82-86   True       mechanism   
7504       ampicillin    drug         49-58   True          effect   
7505       ampicillin    drug       175-184  False  No interaction   
7506       ampicillin    drug       175-184  False  No interaction   
7507      allopurinol    drug         73-83   True          effect   
...               ...     ...           ...    ...             ...   
33503  chlorpropamide    drug       181-194  False  No interaction   
33504  chlorpropamide    drug       181-194  False  No interaction   
33505         alcohol    drug       185-191  False  No interaction   
33506      ethinamate    drug       320-329  False  No interaction   
33507      ethinamate    drug       320-329  False  No interaction   

                                            source_x  
7503   DDICorpus/Train/DrugBank/Estramustine_ddi.xml  
7504    DDICorpus/Train/DrugBank/Clavulanate_ddi.xml  
7505    DDICorpus/Train/DrugBank/Clavulanate_ddi.xml  
7506    DDICorpus/Train/DrugBank/Clavulanate_ddi.xml  
7507    DDICorpus/Train/DrugBank/Clavulanate_ddi.xml  
...                                              ...  
33503    DDICorpus/Train/DrugBank/Lovastatin_ddi.xml  
33504    DDICorpus/Train/DrugBank/Lovastatin_ddi.xml  
33505    DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  
33506    DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  
33507    DDICorpus/Train/DrugBank/Ethinamate_ddi.xml  

[26005 rows x 12 columns]

In [ ]:
len(train_set_drugbank["doc_id"].unique())

511

# Number of Documents in MedLine Training Set

In [ ]:
train_set_medline = df[df["source_x"].str.startswith("DDICorpus/Train/MedLine")]
train_set_medline

doc_id         sentence_id  \
5716  DDI-MedLine.d69  DDI-MedLine.d69.s0   
5717  DDI-MedLine.d69  DDI-MedLine.d69.s0   
5718  DDI-MedLine.d69  DDI-MedLine.d69.s0   
5719  DDI-MedLine.d69  DDI-MedLine.d69.s1   
5720  DDI-MedLine.d69  DDI-MedLine.d69.s1   
...               ...                 ...   
7498  DDI-MedLine.d29  DDI-MedLine.d29.s3   
7499  DDI-MedLine.d29  DDI-MedLine.d29.s3   
7500  DDI-MedLine.d29  DDI-MedLine.d29.s3   
7501  DDI-MedLine.d29  DDI-MedLine.d29.s4   
7502  DDI-MedLine.d29  DDI-MedLine.d29.s8   

                                          sentence_text          e1_text  \
5716  Differential regulation of tyrosine phosphoryl...  contortrostatin   
5717  Differential regulation of tyrosine phosphoryl...  contortrostatin   
5718  Differential regulation of tyrosine phosphoryl...       echistatin   
5719  The homodimeric disintegrin contortrostatin wa...  contortrostatin   
5720  The homodimeric disintegrin contortrostatin wa...  contortrostatin   
...                                                 ...              ...   
7498  To determine whether probenecid has a direct e...      cloxacillin   
7499  To determine whether probenecid has a direct e...      cloxacillin   
7500  To determine whether probenecid has a direct e...      cloxacillin   
7501  No significant difference was found between th...      cloxacillin   
7502  This might be explained by a blockade by probe...       probenecid   

     e1_type e1_charOffset      e2_text e2_type e2_charOffset    ddi  \
5716  drug_n         70-84   echistatin  drug_n       141-150  False   
5717  drug_n         70-84   flavoridin  drug_n       156-165  False   
5718  drug_n       141-150   flavoridin  drug_n       156-165  False   
5719  drug_n         28-42   echistatin  drug_n        96-105  False   
5720  drug_n         28-42   flavoridin  drug_n       111-120  False   
...      ...           ...          ...     ...           ...    ...   
7498    drug         75-85  cloxacillin    drug       124-134  False   
7499    drug         75-85   probenecid    drug       274-283  False   
7500    drug       124-134   probenecid    drug       274-283  False   
7501    drug        93-103   probenecid    drug       122-131  False   
7502    drug         41-50  cloxacillin    drug         74-84   True   

                type                              source_x  
5716  No interaction  DDICorpus/Train/MedLine/10978746.xml  
5717  No interaction  DDICorpus/Train/MedLine/10978746.xml  
5718  No interaction  DDICorpus/Train/MedLine/10978746.xml  
5719  No interaction  DDICorpus/Train/MedLine/10978746.xml  
5720  No interaction  DDICorpus/Train/MedLine/10978746.xml  
...              ...                                   ...  
7498  No interaction  DDICorpus/Train/MedLine/15830476.xml  
7499  No interaction  DDICorpus/Train/MedLine/15830476.xml  
7500  No interaction  DDICorpus/Train/MedLine/15830476.xml  
7501  No interaction  DDICorpus/Train/MedLine/15830476.xml  
7502       mechanism  DDICorpus/Train/MedLine/15830476.xml  

[1787 rows x 12 columns]

In [ ]:
len(train_set_medline["doc_id"].unique())

113

# Number of unique sentences in Training set(in total of 624 documents)

In [ ]:
len(train_set["sentence_id"].unique())

3789

# Number of unique sentences in DrugBank Training set

In [ ]:
len(train_set_drugbank["sentence_id"].unique())

3256

# Number of unique sentences in MedLine Training Set

In [ ]:
len(train_set_medline["sentence_id"].unique())

533

# Drug Distribution in Training Set

In [ ]:
drugbank_e1 = train_set_drugbank["e1_text"].unique()
drugbank_e2 = train_set_drugbank["e2_text"].unique()
drugbank_entities = np.concatenate((drugbank_e1, drugbank_e2))
drugbank_entities

array(['calcium', 'allopurinol', 'ampicillin', ...,
       'HMG-CoA reductase inhibitor class', 'MEVACOR', 'ethinamate'],
      dtype=object)

In [ ]:
len(drugbank_entities)

4097

In [ ]:
entities = pd.concat([
    train_set_drugbank[['e1_text', 'e1_type']].rename(columns={'e1_text': 'entity', 'e1_type': 'label'}),
    train_set_drugbank[['e2_text', 'e2_type']].rename(columns={'e2_text': 'entity', 'e2_type': 'label'})
])
unique_entities = entities.drop_duplicates()

label_distribution = unique_entities['label'].value_counts()
label_distribution

label
drug      1312
group      823
brand      356
drug_n      61
Name: count, dtype: int64

In [ ]:
entities = pd.concat([
    train_set_medline[['e1_text', 'e1_type']].rename(columns={'e1_text': 'entity', 'e1_type': 'label'}),
    train_set_medline[['e2_text', 'e2_type']].rename(columns={'e2_text': 'entity', 'e2_type': 'label'})
])
unique_entities = entities.drop_duplicates()

label_distribution = unique_entities['label'].value_counts()
label_distribution

label
drug      277
drug_n    117
group     101
brand       6
Name: count, dtype: int64

# Interaction Type Distribution in Training Set

In [ ]:
train_set_drugbank["type"].value_counts()

type
No interaction    22217
effect             1535
mechanism          1257
advise              818
int                 178
Name: count, dtype: int64

In [ ]:
train_set_medline["type"].value_counts()

type
No interaction    1555
effect             152
mechanism           62
int                 10
advise               8
Name: count, dtype: int64

# Number of documents in Test set

In [ ]:
test_set = df[df["source_x"].str.startswith("DDICorpus/Test")]
test_set

doc_id           sentence_id  \
0      DDI-MedLine.d180   DDI-MedLine.d180.s1   
1      DDI-MedLine.d180  DDI-MedLine.d180.s10   
2      DDI-MedLine.d180  DDI-MedLine.d180.s10   
3      DDI-MedLine.d180  DDI-MedLine.d180.s10   
4      DDI-MedLine.d180  DDI-MedLine.d180.s10   
...                 ...                   ...   
5711  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5712  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5713  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5714  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5715  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   

                                          sentence_text              e1_text  \
0     The 3-Deazaneplanocin A (DZNep), one of S-aden...  3-Deazaneplanocin A   
1     We also found that Bcl-2 was overexpressed in ...                DZNep   
2     We also found that Bcl-2 was overexpressed in ...                DZNep   
3     We also found that Bcl-2 was overexpressed in ...                DZNep   
4     We also found that Bcl-2 was overexpressed in ...                DZNep   
...                                                 ...                  ...   
5711  SKELAXIN may enhance the effects of alcohol, b...             SKELAXIN   
5712  SKELAXIN may enhance the effects of alcohol, b...             SKELAXIN   
5713  SKELAXIN may enhance the effects of alcohol, b...              alcohol   
5714  SKELAXIN may enhance the effects of alcohol, b...              alcohol   
5715  SKELAXIN may enhance the effects of alcohol, b...         barbiturates   

     e1_type e1_charOffset          e2_text e2_type e2_charOffset    ddi  \
0     drug_n          4-22            DZNep  drug_n         25-29  False   
1     drug_n         46-50            DZNep  drug_n         92-96  False   
2     drug_n         46-50          ABT-737  drug_n       102-108  False   
3     drug_n         46-50            DZNep  drug_n       195-199  False   
4     drug_n         92-96          ABT-737  drug_n       102-108   True   
...      ...           ...              ...     ...           ...    ...   
5711   brand           0-7     barbiturates   group         45-56   True   
5712   brand           0-7  CNS depressants   group         68-82   True   
5713    drug         36-42     barbiturates   group         45-56  False   
5714    drug         36-42  CNS depressants   group         68-82  False   
5715   group         45-56  CNS depressants   group         68-82  False   

                type                                           source_x  
0     No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
1     No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
2     No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
3     No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
4             effect  DDICorpus/Test/Test for DDI Extraction task/Me...  
...              ...                                                ...  
5711          effect  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5712          effect  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5713  No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5714  No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5715  No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  

[5716 rows x 12 columns]

In [ ]:
len(test_set["doc_id"].unique())

191

# Number of documents in DrugBank Test Set

In [ ]:
test_set_drugbank = test_set[test_set["doc_id"].str.startswith("DDI-DrugBank")]
test_set_drugbank

doc_id           sentence_id  \
451   DDI-DrugBank.d765  DDI-DrugBank.d765.s0   
452   DDI-DrugBank.d765  DDI-DrugBank.d765.s1   
453   DDI-DrugBank.d765  DDI-DrugBank.d765.s1   
454   DDI-DrugBank.d765  DDI-DrugBank.d765.s1   
455   DDI-DrugBank.d765  DDI-DrugBank.d765.s1   
...                 ...                   ...   
5711  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5712  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5713  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5714  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   
5715  DDI-DrugBank.d605  DDI-DrugBank.d605.s0   

                                          sentence_text       e1_text e1_type  \
451   CYP3A4 Inhibitors: Ketoconazole, an inhibitor ...  Ketoconazole    drug   
452   For patients receiving ketoconazole or other p...  ketoconazole    drug   
453   For patients receiving ketoconazole or other p...  ketoconazole    drug   
454   For patients receiving ketoconazole or other p...  ketoconazole    drug   
455   For patients receiving ketoconazole or other p...  ketoconazole    drug   
...                                                 ...           ...     ...   
5711  SKELAXIN may enhance the effects of alcohol, b...      SKELAXIN   brand   
5712  SKELAXIN may enhance the effects of alcohol, b...      SKELAXIN   brand   
5713  SKELAXIN may enhance the effects of alcohol, b...       alcohol    drug   
5714  SKELAXIN may enhance the effects of alcohol, b...       alcohol    drug   
5715  SKELAXIN may enhance the effects of alcohol, b...  barbiturates   group   

     e1_charOffset                e2_text e2_type e2_charOffset    ddi  \
451          19-30            tolterodine    drug       135-145   True   
452          23-34      azole antifungals   group        84-100  False   
453          23-34           itraconazole    drug       107-118  False   
454          23-34             miconazole    drug       121-130  False   
455          23-34  macrolide antibiotics   group       136-156  False   
...            ...                    ...     ...           ...    ...   
5711           0-7           barbiturates   group         45-56   True   
5712           0-7        CNS depressants   group         68-82   True   
5713         36-42           barbiturates   group         45-56  False   
5714         36-42        CNS depressants   group         68-82  False   
5715         45-56        CNS depressants   group         68-82  False   

                type                                           source_x  
451        mechanism  DDICorpus/Test/Test for DDI Extraction task/Dr...  
452   No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
453   No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
454   No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
455   No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
...              ...                                                ...  
5711          effect  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5712          effect  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5713  No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5714  No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  
5715  No interaction  DDICorpus/Test/Test for DDI Extraction task/Dr...  

[5265 rows x 12 columns]

In [ ]:
len(test_set_drugbank["doc_id"].unique())

158

# Number of documents in MedLine Test Set

In [ ]:
test_set_medline = test_set[test_set["doc_id"].str.startswith("DDI-MedLine")]
test_set_medline

doc_id           sentence_id  \
0    DDI-MedLine.d180   DDI-MedLine.d180.s1   
1    DDI-MedLine.d180  DDI-MedLine.d180.s10   
2    DDI-MedLine.d180  DDI-MedLine.d180.s10   
3    DDI-MedLine.d180  DDI-MedLine.d180.s10   
4    DDI-MedLine.d180  DDI-MedLine.d180.s10   
..                ...                   ...   
446  DDI-MedLine.d189  DDI-MedLine.d189.s10   
447  DDI-MedLine.d189  DDI-MedLine.d189.s10   
448  DDI-MedLine.d189  DDI-MedLine.d189.s10   
449  DDI-MedLine.d189  DDI-MedLine.d189.s10   
450  DDI-MedLine.d189  DDI-MedLine.d189.s11   

                                         sentence_text              e1_text  \
0    The 3-Deazaneplanocin A (DZNep), one of S-aden...  3-Deazaneplanocin A   
1    We also found that Bcl-2 was overexpressed in ...                DZNep   
2    We also found that Bcl-2 was overexpressed in ...                DZNep   
3    We also found that Bcl-2 was overexpressed in ...                DZNep   
4    We also found that Bcl-2 was overexpressed in ...                DZNep   
..                                                 ...                  ...   
446  Furafylline and sulfaphenazole had no effect, ...          Furafylline   
447  Furafylline and sulfaphenazole had no effect, ...       sulfaphenazole   
448  Furafylline and sulfaphenazole had no effect, ...       sulfaphenazole   
449  Furafylline and sulfaphenazole had no effect, ...            quinidine   
450  These results suggest that rat liver slices do...          precocene I   

    e1_type e1_charOffset      e2_text e2_type e2_charOffset    ddi  \
0    drug_n          4-22        DZNep  drug_n         25-29  False   
1    drug_n         46-50        DZNep  drug_n         92-96  False   
2    drug_n         46-50      ABT-737  drug_n       102-108  False   
3    drug_n         46-50        DZNep  drug_n       195-199  False   
4    drug_n         92-96      ABT-737  drug_n       102-108   True   
..      ...           ...          ...     ...           ...    ...   
446  drug_n          0-10  precocene I  drug_n         82-92  False   
447  drug_n         16-29    quinidine    drug         52-60  False   
448  drug_n         16-29  precocene I  drug_n         82-92  False   
449    drug         52-60  precocene I  drug_n         82-92   True   
450  drug_n       103-113  precocene I  drug_n       165-175  False   

               type                                           source_x  
0    No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
1    No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
2    No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
3    No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
4            effect  DDICorpus/Test/Test for DDI Extraction task/Me...  
..              ...                                                ...  
446  No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
447  No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
448  No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  
449          effect  DDICorpus/Test/Test for DDI Extraction task/Me...  
450  No interaction  DDICorpus/Test/Test for DDI Extraction task/Me...  

[451 rows x 12 columns]

In [ ]:
len(test_set_medline["doc_id"].unique())

33

# Number of unique sentences in test documents (191 documents)

In [ ]:
len(test_set["sentence_id"].unique())

790

# Number of sentences in DrugBank Test Set

In [ ]:
len(test_set_drugbank["sentence_id"].unique())

620

# Number of sentences in MedLine Test Set

In [ ]:
len(test_set_medline["sentence_id"].unique())

170

# Drug distribution in Test Sets

In [ ]:
entities = pd.concat([
    test_set_drugbank[['e1_text', 'e1_type']].rename(columns={'e1_text': 'entity', 'e1_type': 'label'}),
    test_set_drugbank[['e2_text', 'e2_type']].rename(columns={'e2_text': 'entity', 'e2_type': 'label'})
])
unique_entities = entities.drop_duplicates()

label_distribution = unique_entities['label'].value_counts()
label_distribution

label
drug      476
group     304
brand     102
drug_n     14
Name: count, dtype: int64

In [ ]:
entities = pd.concat([
    test_set_medline[['e1_text', 'e1_type']].rename(columns={'e1_text': 'entity', 'e1_type': 'label'}),
    test_set_medline[['e2_text', 'e2_type']].rename(columns={'e2_text': 'entity', 'e2_type': 'label'})
])
unique_entities = entities.drop_duplicates()

label_distribution = unique_entities['label'].value_counts()
label_distribution

label
drug      93
drug_n    33
group     25
brand      9
Name: count, dtype: int64

# Interaction Type Distribution in Test Set

In [ ]:
test_set_drugbank["type"].value_counts()

type
No interaction    4381
effect             298
mechanism          278
advise             214
int                 94
Name: count, dtype: int64

In [ ]:
test_set_medline["type"].value_counts()

type
No interaction    356
effect             62
mechanism          24
advise              7
int                 2
Name: count, dtype: int64